test/user_test.exs

```diff
@@ -0,0 +1,45 @@
+defmodule DoAuth.UserTest do
+  use ExUnit.Case
+  use Bamboo.Test
+
+  alias Uptight.Result
+  alias Uptight.Text, as: T
+  alias DoAuth.User
+
+  describe "reserve_identity/3" do
+    setup do
+      %Result.Ok{ok: email} = T.new("user@mail.com")
+      %Result.Ok{ok: nickname} = T.new("user")
+
+      on_exit(fn -> File.rm_rf!(Path.join(["db", "nonode@nohost"])) end)
+
+      %{email: email, nickname: nickname}
+    end
+
+    test "with unused email allows to reserve identity", %{email: %T{text: email_text} = email, nickname: nickname} do
+      %Result.Ok{ok: hostname} = T.new("do_auth.com")
+      %Result.Ok{ok: pid} = User.reserve_identity(email, nickname, homebase: hostname)
+
+      assert is_pid(pid)
+
+      assert %{
+        email: ^email,
+        nickname: ^nickname,
+        cred: %Uptight.Base.Urlsafe{},
```

Comment:
The problem is currently here: we don't actually encode the `cred` into the desired type and got a plain `map()` instead. The question is how we should do it (encode this resulting map into the desired type).

Reply:
1. Put the credential you generated into the Credential store by calling `Credential.transact_cred/2`.
2. It shall return a credential, which is just a `map()`.
3. It has an `id` in it, which is the same as the `cred["proof"]["signature"]`.
4. We can convert it into a `Urlsafe.t()` and store it in the user state.
5. We can then use `Credential.tip/1` with this `cred`, to retrieve the full map.